Blaise.Tests.Helpers/Instrument/InstrumentHelper.cs

```diff
@@ -15,42 +15,106 @@ public class InstrumentHelper
 
         public InstrumentHelper()
         {
-            _blaiseSurveyApi = new BlaiseSurveyApi();           
+            _blaiseSurveyApi = new BlaiseSurveyApi();
         }
 
         public static InstrumentHelper GetInstance()
         {
             return _currentInstance ?? (_currentInstance = new InstrumentHelper());
         }
 
-        public void InstallInstrument()
+        public SurveyStatusType GetQuestionnaireStatus()
         {
-            _blaiseSurveyApi.InstallSurvey(BlaiseConfigurationHelper.InstrumentName,
-                BlaiseConfigurationHelper.ServerParkName,
-                 BlaiseConfigurationHelper.InstrumentPackage,
-                SurveyInterviewType.Cati);
+            return _blaiseSurveyApi.GetSurveyStatus(BlaiseConfigurationHelper.InstrumentName,
+                BlaiseConfigurationHelper.ServerParkName);
         }
 
-        public void InstallInstrument(string instrumentPackage)
+        private bool DoesSurveyExists(string instrumentName = "")
         {
-            _blaiseSurveyApi.InstallSurvey(BlaiseConfigurationHelper.InstrumentName,
-                BlaiseConfigurationHelper.ServerParkName,
-                instrumentPackage,
-                SurveyInterviewType.Cati);
+            try
+            {
+                return _blaiseSurveyApi.SurveyExists(!string.IsNullOrEmpty(instrumentName) ? instrumentName : BlaiseConfigurationHelper.InstrumentName,
+                    BlaiseConfigurationHelper.ServerParkName);
+            }
+            catch (Nuget.Api.Contracts.Exceptions.DataNotFoundException)
+            {
+                return false;
+            }
+        }
+
+        private bool CheckForErroneousSurvey(string instrumentName = "")
+        {
+            if (_blaiseSurveyApi.GetSurveyStatus(!string.IsNullOrEmpty(instrumentName) ? instrumentName : BlaiseConfigurationHelper.InstrumentName,
+                                                    BlaiseConfigurationHelper.ServerParkName) == SurveyStatusType.Erroneous)
+            {
+                throw new Exception($"ERROR: The DST2111Z questionnaire has failed to delete, current status {Enum.GetName(typeof(SurveyStatusType), SurveyStatusType.Erroneous)} You will probably need to restart Blaise to delete it.");
+            }
+
+            return false;
+        }
+
+        public void InstallInstrument(string instrumentPackage = "")
+        {
+            if (DoesSurveyExists(BlaiseConfigurationHelper.InstrumentName))
+            {
+                _blaiseSurveyApi.UninstallSurvey(BlaiseConfigurationHelper.InstrumentName,
+                    BlaiseConfigurationHelper.ServerParkName);
+                Thread.Sleep(int.Parse(BlaiseConfigurationHelper.UninstallSurveyTimeOutInSeconds) * 1000);
+            }

```

Comment:
Is this sleep for 5*1000 seconds (83 minutes) or have I misread this line?

Reply:
The * 1000 converts the seconds to milliseconds a requirement for the function